extensions/resource-deployment/src/interfaces.ts

```diff
@@ -300,8 +300,8 @@ export interface DynamicOptionsInfo {
 }
 
 export interface DynamicOptionsAlternates {
-	selection: string,
-	alternateValues: string[],
+	selection: string | azdata.CategoryValue,
```

Comment:
Does selection need to allow CategoryValue as well or can we use just the name like we do for defaultValue?

Reply:
I think selection needs to allow CategoryValue because we need to access the .name property if an option is a CategoryValue rather than a string.

Reply:
But shouldn't selection always be the name? Is there any reason we'd need to use the displayName property for the selection?

Reply:
Oh actually you're right-- selection itself does not need to be a CategoryValue, just like the defaultValue. I think though in order to get this working I need to convert this expression to unknown: (<azdata.CategoryValue><unknown>item.selection).name 

Is that alright to do? It gives me an error otherwise.

![image](https://user-images.githubusercontent.com/22386690/153515769-90f5742d-98d9-48af-9470-aa4b3d957b6d.png)

Reply:
Oh hold on, that doesn't seem right since selection should be a string. Let me re-look at this

Reply:
Changed it to be just string-- now it should work